src/Microsoft.TestPlatform.CoreUtilities/Utilities/StringExtensions.cs

```diff
@@ -17,5 +19,28 @@ public static string AddDoubleQuote(this string value)
         {
             return "\"" + value + "\"";
         }
+
+        public static void AppendToStringBuilderBasedOnMaxLength(this string data, StringBuilder result)
```

Comment:
Doesn't look like string extension. May be a string builder extension.
`public static void AppendSafe(this StringBuilder stringBuilder, string data)`

Reply:
Create `StringBuilderExtensions` class with `AppendSafeWithNewLine()`.